src/BookNavigator/volumes/volumes-provider.js

```diff
@@ -17,88 +16,73 @@ export default class VolumesProvider {
    * @param {import('../../BookReader').default} bookreader
    */
   constructor({ baseHost, bookreader, onProviderChange }) {
+    /** @type {import('../../BookReader').default} */
+    this.bookreader = bookreader;
     this.onProviderChange = onProviderChange;
-    this.component = document.createElement("viewable-files");
+    this.baseHost = baseHost;
 
     const files = bookreader.options.multipleBooksList.by_subprefix;
     this.viewableFiles = Object.keys(files).map(item => files[item]);
     this.volumeCount = Object.keys(files).length;
 
-    /** @type {import('../../BookReader').default} */
-    this.bookreader = bookreader;
-
-    this.component.subPrefix = bookreader.options.subPrefix || "";
-    this.component.hostUrl = baseHost;
-    this.component.viewableFiles = this.viewableFiles;
-
     this.id = "volumes";
     this.label = `Viewable files (${this.volumeCount})`;
-    this.icon = html`${volumesIcon}`;
+    this.icon = html`${viewableFilesIcon}`;
+    this.sortOrderBy = sortTypes.default;
 
-    this.sortOrderBy = sortType.default;
+    this.component = document.createElement("iaux-viewable-files");
+    this.component.addSortToUrl = true;
+    this.component.subPrefix = bookreader.options.subPrefix || "";
+    this.component.baseHost = baseHost;
+    this.component.fileList = [...this.viewableFiles];
+
+    this.sortFilesComponent = document.createElement("iaux-sort-viewable-files");
```

Comment:
thought about it. right now,  viewable-files panel does not have access to the header of item nav, this piece must be passed through via config.  refactoring for otherwise is not in scope at the moment.